pkg/command/command.go

```diff
@@ -7,100 +7,44 @@ import (
-
-func Run(args []string) (string, error) {
 	defaultPath, err := os.Getwd()
 	if err != nil {
 		return "", err
 	}
-
-	newCommand := flag.NewFlagSet("new", flag.ExitOnError)
-	newToken := newCommand.String("token", "", "Access token (PAT) to git provider")
-	newGroup := newCommand.String("group", "", "Main application group")
-	newApp := newCommand.String("app", "", "Name of the application")
-	newTag := newCommand.String("tag", "", "Application version/tag to set")
-	newProviderType := newCommand.String("provider", "azdo", "git provider to use")
-	newPath := newCommand.String("sourcedir", defaultPath, "Source working tree to operate on")
-
-	promoteCommand := flag.NewFlagSet("promote", flag.ExitOnError)
-	promoteToken := promoteCommand.String("token", "", "Access token (PAT) to git provider")
-	promoteProviderType := promoteCommand.String("provider", "azdo", "git provider to use")
-	promotePath := promoteCommand.String("sourcedir", defaultPath, "Source working tree to operate on")
-
-	statusCommand := flag.NewFlagSet("status", flag.ExitOnError)
-	statusToken := statusCommand.String("token", "", "Access token (PAT) to git provider")
-	statusProviderType := statusCommand.String("provider", "azdo", "git provider to use")
-	statusPath := statusCommand.String("sourcedir", defaultPath, "Source working tree to operate on")
-
-	if len(args) < 2 {
-		return "", fmt.Errorf("new, promote or status subcommand is required")
+	global := flag.NewFlagSet(args[1], flag.ExitOnError)
+	token := global.String("token", "", "Access token (PAT) to git provider")
+	providerType := global.String("provider", "azdo", "The git provider to use")
+	path := global.String("sourcedir", defaultPath, "Source working tree to operate on")
+	err = global.Parse(args[2:])
+	if err != nil {
+		return "", err
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	var commandErr error
-	var message string
-	switch args[1] {
-	case "new":
-		err := newCommand.Parse(args[2:])
-		if err != nil {
-			return "", err
-		}
-		message, commandErr = withCopyOfWorkTree(newPath, func(workTreeCopy string) (string, error) {
-			return NewCommand(ctx, *newProviderType, workTreeCopy, *newToken, *newGroup, *newApp, *newTag)
-		})
-	case "promote":
-		err := promoteCommand.Parse(args[2:])
-		if err != nil {
-			return "", err
-		}
-		message, commandErr = withCopyOfWorkTree(promotePath, func(workTreeCopy string) (string, error) {
-			return PromoteCommand(ctx, *promoteProviderType, workTreeCopy, *promoteToken)
-		})
-	case "status":
-		err := statusCommand.Parse(args[2:])
-		if err != nil {
-			return "", err
-		}
-		message, commandErr = withCopyOfWorkTree(statusPath, func(workTreeCopy string) (string, error) {
-			return StatusCommand(ctx, *statusProviderType, workTreeCopy, *statusToken)
-		})
-	default:
-		flag.PrintDefaults()
-		return "", fmt.Errorf("Unknown flag: %s", args[1])
+	// Load configuration
+	file, err := os.Open(fmt.Sprintf("%s/%s", *path, configFileName))
```

Comment:
As long as we're changing this, can we use filepath to construct the filename? It is not that I expect this to run on Windows any time soon, it is just that when I read the code, I know that I don't have to worry about trailing slashes and other path-specific corner cases.